scvi/nn/_base_components.py

```diff
@@ -340,6 +351,7 @@ def __init__(
         inject_covariates: bool = True,
         use_batch_norm: bool = False,
         use_layer_norm: bool = False,
+        batch_embedding: nn.Embedding = None,
```

Comment:
Little odd to me to take an Embedding object here.  Injecting external state into the constructor is inconsistent with the rest of the parametrization. Would prefer you convert to embedding in the `generative` function and then pass it in as a vector to the decoder. Then just make sure the decoder can accept this additional input by replacing this parameter with a int (similar to what `n_cat_list` does).